Add USwitch player to OVS

mbertrand opened the pull request:
#### What are the relevant tickets?
Fixes #93

#### What's this PR do?
- Adds a boolean field to the Video model ('multiangle')
- if the video is multiangle, loads the USwitch player instead of the video.js player

#### How should this be manually tested?
- Add this line to your `.env` file: `USWITCH_URL=https://s3.amazonaws.com/odl-video-service-uswitch-dev`
- `docker-compose build`
- `docker-compose run`
- Using the admin console, set 'multiangle=True' on any video
- Modify your `/etc/hosts` so that one of the following domains (`localhost`, `mit.edu`, `amps-web.amps.ms.mit.edu`) points to your docker host, for example `192.168.99.100 amps-web.amps.ms.mit.edu`
- Load the video detail page for the modified video, using `amps-web.amps.ms.mit.edu` as the domain.  You should see the USwitch player instead of the video.js player.
- Play the video, there will be one main window and 4 smaller windows at the top showing the 4 corners of the video, labeled Camera 1-4.

#### Any background context you want to provide?
There is an issue with styling on the detail page when the USwitch player is used - #104 
The player only works with certain domains: #108
This assumes the USwitch JS/CSS will be hosted on S3: #103

noisecapella commented:
When I click on the fullscreen icon for this video player it makes the browser full screen, not the video itself. Is there some configuration setting relevant to this?

noisecapella commented:
When I switch from the multivideo view it pops up a window but the URL is not found:

![screenshot from 2017-08-02 10-50-59](https://user-images.githubusercontent.com/863262/28879570-7d0f4e68-7770-11e7-83b2-198f29f5e55a.png)

noisecapella commented on static/scss/layout.scss:
```diff
@@ -25,3 +26,14 @@
     bottom: 0;
 }
 
+// OmniPlayer styles
```
At some point we should add a CSS linter, but in the mean time can you make the indentation consistent below?

noisecapella commented:
The player seems to absorb all keystrokes. If you click on the description textbox for example and type something, no text appears

noisecapella commented on static/js/entry/uswitch.js:
```diff
@@ -0,0 +1,26 @@
+/* global SETTINGS: false */
+require('react-hot-loader/patch');
+const $script = require('scriptjs');
```
I'm not sure we should use `$script` here. The below async imports depend on one another so that they're loaded in sequence instead of in parallel. If we include the script tags in the HTML right before the `render_bundle` for this entry we should get the same behavior and the script tags will load in parallel too

noisecapella commented on ui/views.py:
```diff
@@ -85,15 +85,24 @@ class VideoDetail(DetailView):
+                default_settings["uswitchPlayerURL"] = settings.USWITCH_URL
         except VideoFile.DoesNotExist:
             videofile = None
+            default_settings['videofile'] = ''
```
`videofile` is an object above but it's a string here. Can you make this `None` instead?

noisecapella commented on ui/views_test.py:
```diff
@@ -68,3 +72,16 @@ def test_video_detail_unencoded(
+    video.save()
+    url = reverse('video-detail', kwargs={'pk': video.id})
+    response = admin_client.get(url)
+    assert 'uswitchPlayerURL' in response.context_data
```
Can you assert the value in the context here?

noisecapella replied:
We should also set a fake value for `settings.USWITCH_URL` so that we know that it is the source of the value here. By default it's an empty string

mbertrand commented:
Looks like the USwitch player needs to be run in an iframe to avoid problems listed above (fullscreen mode not working correctly, text input absorbed), yuck.  Made that change.  Fixed the issue with the `mosaic.html` popup window by creating a new view and template for it, where the template contains the `div` tag from the original `mosaic.html` supplied with the player, and the script from that original HTML has been moved to S3 in a new file [mosaic.js](https://github.com/mitodl/omniplayer/blob/master/mosaic.js)

noisecapella commented on odl_video/test_settings.py:
```diff
@@ -0,0 +1,5 @@
+Django test settings for odl_video project.
+"""
+from odl_video.settings import *  # pylint: disable=wildcard-import,unused-wildcard-import
+USWITCH_URL = 'https://testing_odl.mit.edu'
```
Instead of setting the URL here can we set it only in the test where we need it set? The `settings` fixture should let you modify settings on a per test basis

mbertrand replied:
Do you have an example of that? I originally tried the following in `conftest.py`:
```
def pytest_configure():
    settings.configure(USWITCH_URL='https://testing_odl.mit.edu')
```

but then all the tests failed because 'settings were already configured', in pytest.ini: `ds=odl_video.settings`

mbertrand replied:
Made the change

noisecapella commented:
Yes, all the functionality issues are fixed now

noisecapella commented:
Can you also revert the change to `yarn.lock` since none of the packages are changing anymore?

mbertrand commented:
I removed `scriptjs` from yarn (the only new thing I had added), `videojs-contrib-quality-levels` should have been removed earlier (not being used), not sure what triggered the other changes in `yarn.lock`
